src/benchmarks/micro/libraries/System.Utf8String.Experimental/Perf.Utf8String.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using ustring = System.Utf8String;
+
+namespace System.Text.Experimental
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.Runtime)]
+    public class Perf
+    {
+        public static string ascii_11;
+        public static string nonascii_110;
+        public static string nonascii_chinese;
+        public static string nonascii_cyrillic;
+        public static string nonascii_greek;
+        public static IEnumerable<string> NonAsciiData()
+        {
+            yield return nonascii_110;
+            yield return nonascii_chinese;
+            yield return nonascii_cyrillic;
+            yield return nonascii_greek;
+        }
+
+        [GlobalSetup]
+        public void InitializeData()
+        {
+            ascii_11 = File.ReadAllText("../../../../../src/benchmarks/micro/libraries/System.Utf8String.Experimental/11.txt");
+            nonascii_110 = File.ReadAllText("../../../../../src/benchmarks/micro/libraries/System.Utf8String.Experimental/11-0.txt");
+            nonascii_chinese = File.ReadAllText("../../../../../src/benchmarks/micro/libraries/System.Utf8String.Experimental/25249-0.txt");
+            nonascii_cyrillic = File.ReadAllText("../../../../../src/benchmarks/micro/libraries/System.Utf8String.Experimental/30774-0.txt");
+            nonascii_greek = File.ReadAllText("../../../../../src/benchmarks/micro/libraries/System.Utf8String.Experimental/39251-0.txt");
+        }
+
+        public static IEnumerable<string> AsciiData()
+        {
+            yield return ascii_11;
+        }
+
+        [Benchmark]
+        [ArgumentsSource(nameof(NonAsciiData))]
+        public int ToUtf16(string expected)
+        {
+            Utf8Span span = new Utf8Span(new Utf8String(expected));
```

Comment:
Don't call `new Utf8String(string)` in these benchmarks, as it performs a considerable amount of work and will throw off your numbers. Instead, have a benchmark dedicated just to calling this method. For all other benchmarks, use the `[GlobalSetup]` routine to pre-populate an instance field of type `Utf8String`.

In your `[GlobalSetup]` routine, you should check whether the first char of each string is `'\uFEFF'`, and if so then strip that character off the front of the string (or just throw an exception from within your setup routine if you detect this). That will work around any stray BOM issues that might throw off the benchmark.

Reply:
That's the plan with `GlobalSetup`. This is still WIP. I'm hitting an exception that throws "System.ComponentModel.Win32Exception: 'The filename or extension is too long'" though, so I'm still trying to debug that. 

Yup, I'll strip out the BOM